ResearchKit/ActiveTasks/ORKTowerOfHanoiViewController.m

```diff
@@ -0,0 +1,256 @@
+    [super viewWillDisappear:animated];
+    [_timer invalidate];
+}
+
+- (void)updateViewConstraints {
+    [super updateViewConstraints];
+    if (_currentConstraints) {
+        [NSLayoutConstraint deactivateConstraints:_currentConstraints];
+        _currentConstraints = nil;
+    }
+    BOOL needCompactLayout = self.traitCollection.horizontalSizeClass == UIUserInterfaceSizeClassCompact &&
+    self.traitCollection.verticalSizeClass != UIUserInterfaceSizeClassCompact;
+    _currentConstraints = needCompactLayout ? [self compactConstraints] : [self regularConstraints];
+    [NSLayoutConstraint activateConstraints:_currentConstraints];
+}
+
+#pragma Mark -- ORKStepViewController
+
+- (void)skipForward {
+    [self finish];
+}
+
+#pragma Mark -- ORKActiveTaskViewController
+
+- (ORKResult *)result {
+    ORKTowerOfHanoiResult *result = [[ORKTowerOfHanoiResult alloc] initWithIdentifier:self.step.identifier];
+    result.numberOfMoves = _numberOfMoves;
+    result.puzzleWasSolved = [self puzzleIsSolved];
+    if (_firstMoveDate != nil) {
+        result.startDate = _firstMoveDate;
+    }
+    return result;
+}
+
+#pragma Mark -- ORKTowerOfHanoiTowerViewDataSource
+ 
+- (NSInteger)numberOfDisksInTowerOfHanoiView:(ORKTowerOfHanoiTowerView *)towerView {
+     NSInteger towerIndex = [_towerViews indexOfObject:towerView];
+     ORKTowerOfHanoiTower *tower = _towers[towerIndex];
+     return tower.disks.count;
+}
+ 
+- (NSInteger)towerOfHanoiView:(ORKTowerOfHanoiTowerView *)towerView sizeForDiskAtIndex:(NSInteger)index {
+     NSInteger towerIndex = [_towerViews indexOfObject:towerView];
+     ORKTowerOfHanoiTower *tower = _towers[towerIndex];
+     return [tower.disks[index]integerValue];
+}
+
+#pragma Mark -- ORKTowerOfHanoiTowerViewDelegate
+
+- (void)towerOfHanoiTowerViewWasSelected:(ORKTowerOfHanoiTowerView *)towerView {
+    NSInteger newSelectedIndex = [_towerViews indexOfObject:towerView];
+    if (_selectedIndex == nil) {
+        _selectedIndex = @(newSelectedIndex);
+    }
+    else if ([_selectedIndex isEqual: @(newSelectedIndex)]) {
+        _selectedIndex = nil;
+    }
+    else {
+        [self transferDiskFromTowerAtIndex:_selectedIndex.integerValue toTowerAtIndex:newSelectedIndex];
+    }
+    [self reloadData];
+    [self evaluatePuzzle];
+}
+
+#pragma Mark -- ORKTowerOfHanoiViewController
+
+- (void)updateTitleText {
+    NSString *text = [NSString stringWithFormat: ORKLocalizedString(@"TOWER_OF_HANOI_TASK_ACTIVE_STEP_PROGRESS_TEXT", nil), (int)_numberOfMoves, [self timeElapsedString]];
+    [self.activeStepView updateTitle:nil text: text];
+}
+
+- (NSString *)timeElapsedString {
+    NSString *seconds = [NSString stringWithFormat:@"%02d", (int)_secondsElapsed % 60];
+    NSString *minutes = [NSString stringWithFormat:@"%02d", (int)(_secondsElapsed / 60) % 60];
+    NSString *hours = [NSString stringWithFormat:@"%02d",(int)_secondsElapsed / 3600];
+    NSString *time = [NSString stringWithFormat:@"%@:%@", minutes,seconds];
```

Comment:
Thanks for the link. My understanding of the doc is that the intended use of NSDateIntervalFormatter is to output the actual dates in a range ie: 12:31:10PM - 12:30:20PM, rather than the interval between the dates as a string ie: 01:20 (1 min, 20 secs) ?